conan/internal/cache/db/table.py

```diff
@@ -10,20 +11,23 @@ class BaseDbTable:
     row_type: namedtuple = None
     columns: namedtuple = None
     unique_together: tuple = None
+    _connection_mutex: threading.Lock = None
 
     def __init__(self, filename):
         self.filename = filename
         column_names: List[str] = [it[0] for it in self.columns_description]
         self.row_type = namedtuple('_', column_names)
         self.columns = self.row_type(*column_names)
+        self._connection_mutex = threading.Lock()
```

Comment:
I am not sure this would cover all cases.
Assigning a value makes it an object attribute, not a class attribute. And the ``DBTable`` object (its children RecipesDBTable and PackagesDBTable) will be instantiated in some scenarios inside the threads, like the parallelism of the ``conan download command``. Probably it should be created as a class member.

Reply:
Is one class-level lock enough, or better make it `_db_locks = defaultdict(threading.Lock)`? In first case we will prohibit parallel access to different databases.

Reply:
Yeah, at a given ``conan`` execution, there will be 2 DBs, the "packages" one, and the "users" one. This lock will be global for the "packages" DB, as the "users" one do not use this code at all. There is only 1 "packages" DB at execution time, so it shouldn't be an issue, a class-level should be enough if I am not missing anything.

Reply:
Referencing it via `type(self)` to be extra sure that we use class attribute and not instance one.